src/utilities/__tests__/getOperationRootType-test.js

```diff
@@ -0,0 +1,136 @@
+ */
+
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import { getOperationRootType } from '../getOperationRootType';
+import { parse, GraphQLSchema, GraphQLObjectType, GraphQLString } from '../../';
+
+const queryType = new GraphQLObjectType({
+  name: 'FooQuery',
+  fields: () => ({
+    field: { type: GraphQLString },
+  }),
+});
+
+const mutationType = new GraphQLObjectType({
+  name: 'FooMutation',
+  fields: () => ({
+    field: { type: GraphQLString },
+  }),
+});
+
+const subscriptionType = new GraphQLObjectType({
+  name: 'FooSubscription',
+  fields: () => ({
+    field: { type: GraphQLString },
+  }),
+});
+
+describe('getOperationRootType', () => {
+  it('Gets a Query type for an unnamed OperationDefinitionNode', () => {
+    const testSchema = new GraphQLSchema({
+      query: queryType,
+    });
+    const doc = parse('{ field }');
+    expect(getOperationRootType(testSchema, doc.definitions[0])).to.equal(
+      queryType,
+    );
+  });
+
+  it('Gets a Query type for an named OperationDefinitionNode', () => {
+    const testSchema = new GraphQLSchema({
+      query: queryType,
+    });
+
+    const doc = parse('query Q { field }');
+    expect(getOperationRootType(testSchema, doc.definitions[0])).to.equal(
+      queryType,
+    );
+  });
+
+  it('Gets a type for OperationTypeDefinitionNodes', () => {
+    const testSchema = new GraphQLSchema({
+      query: queryType,
+      mutation: mutationType,
+      subscription: subscriptionType,
+    });
+
+    const doc = parse(
+      'schema { query: FooQuery mutation: FooMutation subscription: FooSubscription }',
+    );
+    const operationTypes = doc.definitions[0].operationTypes;
+    expect(getOperationRootType(testSchema, operationTypes[0])).to.equal(
+      queryType,
+    );
+    expect(getOperationRootType(testSchema, operationTypes[1])).to.equal(
+      mutationType,
+    );
+    expect(getOperationRootType(testSchema, operationTypes[2])).to.equal(
+      subscriptionType,
+    );
+  });
+
+  it('Gets a Mutation type for an OperationDefinitionNode', () => {
+    const testSchema = new GraphQLSchema({
+      mutation: mutationType,
+    });
+
+    const doc = parse('mutation { field }');
+    expect(getOperationRootType(testSchema, doc.definitions[0])).to.equal(
+      mutationType,
+    );
+  });
+
+  it('Gets a Subscription type for an OperationDefinitionNode', () => {
+    const testSchema = new GraphQLSchema({
+      subscription: subscriptionType,
+    });
+
+    const doc = parse('subscription { field }');
+    expect(getOperationRootType(testSchema, doc.definitions[0])).to.equal(
+      subscriptionType,
+    );
+  });
+
+  it('Throws when query type not defined in schema', () => {
+    const testSchema = new GraphQLSchema({});
+
+    const doc = parse('query { field }');
+    expect(() => getOperationRootType(testSchema, doc.definitions[0])).to.throw(
+      'Schema does not define the required query root type.',
+    );
```

Comment:
That's a good suggestion, though I'm having a hard time finding other examples of this being done. I'm not really familiar with Chai, so I'm OK not being the first person to use all its best features. Additionally, this feels closer to "testing the current API and internal structure for GraphQLError" instead of "testing getOperationRootType". Basically, if we make a change to one of GraphQLError's properties, I don't want the person doing that to have to figure out why this test is failing.

Reply:
@mjmahone You right I run 'grep' and there are no such checks in other places.
I still think it's important to test that errors are bind to correct location.
But you right such tests shouldn't be very dependent on `GraphQLError` 👍